### PR TITLE
Fix license versions for semver and typo

### LINKED ISSUE #2488: Start using 3.24.0 and 3.25.0 for listVersionAdded
After the 3.24.0 release goes out, we should:

1. consider updating "3.24" to "3.24.0" for licenses with `listVersionAdded="3.24.0"`
2. start using "3.25.0" for future `listVersionAdded` licenses
3. review documentation to update any references to version numbers / listVersionAdded for 3.24.0 and later
4. update release-process docs to also mention updating RELEASE-HISTORY.md prior to release
5. fix typo "comparision" => "comparison" in RELEASE-NOTES.md (b/c now I'm just seeing other release-related things and want to track them for the next one)  :)

### PR DESCRIPTION
**Pull request by swinslow**
Fixes #2488

This commit adds ".0" for many licenses with listVersionAdded 3.24 (and a couple with 3.25) following the move to semver for the 3.24.0 release.

It also fixes a long-standing and much-copied typo in the release notes that was bothering me.

This handles items 1, 2 and 5 from #2488 (@xsuchy already took care of 3 and 4 -- thank you!)

Signed-off-by: Steve Winslow <steve@swinslow.net>

### CODE REVIEW

**Comment by swinslow** [comment]
@jlovejoy after the tests pass here, could you merge this if all looks good? Thanks!

**Review by xsuchy — APPROVED** [review]

**Pull request merged**
